pkg/v1/tkg/client/management_components.go

```diff
@@ -66,16 +87,24 @@ func (c *TkgClient) InstallManagementComponents(kubeconfig, kubecontext string)
 		os.Remove(kappControllerConfigFile)
 	}
 
-	return nil
+	return err
 }
 
-func (c *TkgClient) getTKGPackageConfigValuesFile() (string, error) {
-	userProviderConfigValues, err := c.getUserConfigVariableValueMap()
+func (c *TkgClient) getTKGPackageConfigValuesFile(managementPackageVersion, kubeconfig, kubecontext string, upgrade bool) (string, error) {
+	var userProviderConfigValues map[string]string
+	var err error
+
+	if upgrade {
+		userProviderConfigValues, err = c.getUserConfigVariableValueMapForUpgrade(kubeconfig, kubecontext)
```

Comment:
Question: What are the user provided variables that we expect for mgmt cluster packages?

Reply:
It will be all the configuration variables that user provides to create management cluster. Some of those values will be used as a default values as part of clusterclass deployment.